plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorTest.java

```diff
@@ -70,18 +72,47 @@ public void tearDown() throws CoreException {
   @Test
   public void testValidate_withWrongVersion()
       throws IOException, CoreException, ParserConfigurationException {
-    byte[] bytes = XML.getBytes(StandardCharsets.UTF_8);
-    WebXmlValidator validator = new WebXmlValidator();
+    String xml = "<web-app xmlns='http://xmlns.jcp.org/xml/ns/javaee'"
+        + " version='3.1'></web-app>";
+    byte[] bytes = xml.getBytes(StandardCharsets.UTF_8);
     validator.validate(resource, bytes);
     IMarker[] markers = resource.findMarkers(SERVLET_MARKER, true, IResource.DEPTH_ZERO);
     assertEquals(1, markers.length);
     assertEquals(ELEMENT_MESSAGE, (String) markers[0].getAttribute(IMarker.MESSAGE));
   }
   
+  @Test
+  public void testValidate_withExistingServletClass()
+      throws IOException, CoreException, ParserConfigurationException {
+    String xml = "<web-app xmlns='http://java.sun.com/xml/ns/javaee' version='2.5'>"
+        + "<servlet-class>ServletClass</servlet-class>"
+        + "</web-app>";
+    byte[] bytes = xml.getBytes(StandardCharsets.UTF_8);
+    validator.validate(resource, bytes);
+    IMarker[] markers = resource.findMarkers(UNDEFINED_SERVLET_MARKER, true, IResource.DEPTH_ZERO);
+    assertEquals(0, markers.length);
+  }
+  
+  @Test
+  public void testValidate_nonexistingServletClass()
+      throws IOException, CoreException, ParserConfigurationException {
+    String xml = "<web-app xmlns='http://java.sun.com/xml/ns/javaee' version='2.5'>"
+        + "<servlet-class>DoesNotExist</servlet-class>"
+        + "</web-app>";
+    byte[] bytes = xml.getBytes(StandardCharsets.UTF_8);
+    validator.validate(resource, bytes);
+    IMarker[] markers = resource.findMarkers(UNDEFINED_SERVLET_MARKER, true, IResource.DEPTH_ZERO);
+    assertEquals(1, markers.length);
+    String expectedMessage = "DoesNotExist " + Messages.getString("undefined.servlet.class");
+    assertEquals(expectedMessage, markers[0].getAttribute(IMarker.MESSAGE));
+  }
+  
   private static void createFolders(IContainer parent, IPath path) throws CoreException {
```

Comment:
We already have similar methods in `com.google.cloud.tools.eclipse.appengine.newproject.CodeTemplates`. We could extract it into a util class and use it here instead of reimplementing. In a separate PR.